src/pulseaudio_dbus/init.lua

```diff
@@ -349,6 +349,39 @@ function pulse.Device:toggle_muted()
   return self:is_muted()
 end
 
+--- Get the current active port object path
+-- @return the active port object path
+-- @return userdata If the device doesn't have any ports. Userdata being GDBus.Error:org.PulseAudio.Core1.NoSuchPropertyError.
+-- @see pulse.Device:set_active_port
+function pulse.Device:get_active_port()
+  return self:Get("org.PulseAudio.Core1.Device", "ActivePort")
+end
+
+--- Set the active port object path
+-- @tparam string value port object path
+-- @return false If the value it's not a valid port in this device.
+-- @see pulse.Device:get_active_port
+function pulse.Device:set_active_port(value)
+  local available_ports = self:Get("org.PulseAudio.Core1.Device", "Ports")
+  local port_is_valid
+  for _, p in ipairs(available_ports) do
+    if p == value then
+      port_is_valid = true
+      break
+    end
+  end
+
+  if port_is_valid then
+    self:Set("org.PulseAudio.Core1.Device",
+             "ActivePort",
+             lgi.GLib.Variant("o", value))
+    self.ActivePort = {signature="o", value=value}
+  else
+    -- value it's not a valid port in this device
+    return false
```

Comment:
I think that we should throw an error rather than returning a boolean. We could simply do something like

``` lua
assert(port_is_valid, string.format("Invalid port (%s) for device %s", value, self.Name))
```

(and update the docstring) on line 374 rather than `if..else`

We should also add a test please.

Reply:
No problem, this actually makes a lot of sense. I don't know what exactly you wanna test for though. If an invalid port fails?

Reply:
>I don't know what exactly you wanna test for though.

I think that there should be a test for both a valid and an invalid port. I think we already cover the case for a valid port, so we can simply add a test that tries to set a bogus port and detects the error.

The [Busted Test Framework](http://olivinelabs.com/busted/) has `assert.has_error` which should be helpful.

Reply:
Thanks for the tip, the test is pretty minimal but I think it should be enough.